Upgrade to latest version of Asciidoctor

Buzzardo opened the pull request:
Upgrade to latest version of Asciidoctor because we were several
versions behind and were steadily incurring more technical debt.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #).

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc).
-->

Commit:
Upgrade to latest version of Asciidoctor

Upgrade to latest version of Asciidoctor because we were several
versions behind and were steadily incurring more technical debt.

artembilan commented on src/reference/asciidoc/index-header.adoc:
```diff
@@ -3,7 +3,7 @@
 Mark Fisher; Marius Bogoevici; Iwein Fuld; Jonas Partner; Oleg Zhurakousky; Gary Russell; Dave Syer; Josh Long; David Turanski; Gunnar Hillert; Artem Bilan; Amol Nayak; Jay Bryant
 
 ifdef::backend-html5[]
-*{project-version}*
```
So, looks like `project-version` has gone.

Please, revise all our docs then to replace this one with a new `revnumber`.

For example right now now I see this:
```
<version>{project-version}</version>
```
Or this `compile "org.springframework.integration:spring-integration-amqp:{project-version}"`

Has to be an actual project version we built these docs against.

Buzzardo replied:
I added a project-version attribute back into the asciidoctor and asciidoctorPdf tasks. That fixed it.

artembilan reviewed (changes requested):
Just only one version placeholder concern! 👍

artembilan commented on build.gradle:
```diff
@@ -2,22 +2,25 @@ buildscript {
 	repositories {
 		maven { url 'https://repo.spring.io/plugins-release' }
+		jcenter()
+		mavenCentral()
```
Sorry, forgot to ask: do we really need these extra repos?

Buzzardo replied:
Yes, we do. I tried it without them, but it doesn't work. Those repos are where the asciidoctor and asciidoctorPdf tasks come from, so the build breaks without them. I share your pain here: It seems goofy to add those.

Commit:
Replace project-version attribute in Asciidoctor

I had removed the project-version attribute, but the docs use it all
over, so I put it back.

Commit:
Removed unneeded repositories

It turns out that those repositories are not needed.

artembilan commented on build.gradle:
```diff
@@ -6,18 +6,19 @@ buildscript {
 	id 'org.jetbrains.dokka' version '0.10.1'
+	id 'org.asciidoctor.jvm.pdf' version '3.1.0'
+	id 'org.asciidoctor.jvm.gems' version '3.1.0'
+	id 'org.asciidoctor.jvm.convert' version '3.1.0'
```
Can we extract this as a `asciidoctorVersion`?
Alongside with that `ext.kotlinVersion = '1.3.71'` above.

Thanks

Buzzardo replied:
I don't think the asciidoctor folks are trying to keep them in sync. I think it's a coincidence that they are the same just now. We can do it, but we may end up with three variables down the road. I'll go ahead and do it, though, since it seems to be Spring's usual way of doing things.

artembilan reviewed (changes requested):
@Buzzardo , while we are here, would you mind to fix a `mongodb.adoc` for the `reactive-streams.adoc/reactive-message-handler` typo? Has to be `reactive-streams.adoc#reactive-message-handler`.

Nice to have this upgrade: it has caught this issue! 😄

Commit:
Use a variable for the asciidoctor versions

Use a variable to specify the version of the various asciidoctor
components. If they do not remain in sync, we may need multiple
variables in the future. Also, the Gradle documentation says to
set this up across three files.

artembilan commented on gradle.properties:
```diff
@@ -2,3 +2,4 @@ version=5.3.0.BUILD-SNAPSHOT
 org.gradle.jvmargs=-Xmx1536m -Dfile.encoding=UTF-8
 org.gradle.caching=true
 org.gradle.parallel=true
+asciidoctorVersion=3.1.0
```
No, I kinda agree with you about different versions in the future.

Also I don't like the change spreaded to several files which are fully not related to docs.
Can we, please, revert this commit?

Buzzardo replied:
I didn't like it rolling into three files, either. I'll revert it.

Commit:
Fix a broken cross-reference

A cross-reference was incorrectly formatted (missing the # character).

artembilan commented on src/reference/asciidoc/mongodb.adoc:
```diff
@@ -548,7 +548,7 @@ Starting with version 5.3, the `ReactiveMongoDbStoringMessageHandler` and `React
 
 The `ReactiveMongoDbStoringMessageHandler` is an implementation of the `ReactiveMessageHandler` which is supported natively in the framework when reactive streams composition is involved in the integration flow definition.
-See more information in the <<./reactive-streams.adoc/reactive-message-handler,ReactiveMessageHandler>>.
+See more information in the <<./reactive-streams.adoc/#reactive-message-handler,ReactiveMessageHandler>>.
```
I think this is still wrong.
Must be:
```suggestion
See more information in the <<./reactive-streams.adoc#reactive-message-handler,ReactiveMessageHandler>>.
```

Buzzardo replied:
Got it this time.

Commit:
Revert "Use a variable for the asciidoctor versions"

This reverts commit 079cd9bd5381363cbeaa0ba7cbf2d065a321cf3a.

Commit:
Fixing a bad reference (again)

I didn't realize the / character was also a problem.
